tools/dochack/dochack.nim

```diff
@@ -2,16 +2,17 @@ import dom
 import fuzzysearch
 
 
-proc switchTheme(event: Event) =
-  if event.target.checked:
-    document.documentElement.setAttribute("data-theme", "dark")
-    window.localStorage.setItem("theme", "dark")
-  else:
-    document.documentElement.setAttribute("data-theme", "light")
-    window.localStorage.setItem("theme", "light")
+proc setTheme(theme: cstring) {.exportC.} =
+  document.documentElement.setAttribute("data-theme", theme)
+  window.localStorage.setItem("theme", theme)
+
+# set `data-theme` attribute early to prevent white flash
+setTheme((let t = window.localStorage.getItem("theme"); if t.isNil: cstring"auto" else: t))
```

Comment:
Note that this still adds `t` to the global scope. You can use `block` or make it a proc and immediately call it.

Reply:
does it matter ? nim exposes other symbols aswell.
![image](https://user-images.githubusercontent.com/33235928/184546222-96495dfe-44a3-4734-a5a9-f1d5e83ca42a.png)

Reply:
No, it doesn't matter.